### PR TITLE
Add sysroles_fingerprint spec and parser for RHEL System Roles telemetry

### PR DESCRIPTION
**Pull request by sabhasin**
Check all that apply:

- [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
- [x] No Sensitive Data in this change?
- [ ] Is this PR to correct an issue?
- [x] Is this PR an enhancement?

## Description

This PR adds a spec and parser to collect and parse `/var/log/sysroles.jsonl` containing RHEL System Roles execution fingerprints, enabling downstream extraction rules to process role telemetry data.

Background

The rhel-system-roles package generates execution fingerprints in /var/log/sysroles.jsonl in JSONL format (newline-delimited JSON). Each line contains a JSON object with telemetry about a system role execution including:
- role_name: Fully qualified role name (e.g., redhat.rhel_system_roles.network)
- role_path: Path to the role on the system
- status: Execution status (success, failed, begin)
- ansible_version: Version of Ansible used
- managed_node_distro: Managed node distribution (e.g., RedHat-9.4)
- play_hosts_number: Number of hosts in the play
- ansible_check_mode: Whether check mode was enabled
- date: Timestamp of execution

This data is needed for insights analysis and extraction rules to understand system role usage patterns and troubleshoot role execution issues.

##Changes Made

Spec Definition:
- Added sysroles_fingerprint RegistryPoint to insights/specs/__init__.py
- Implemented spec in insights/specs/default.py using simple_file("/var/log/sysroles.jsonl")

Parser Implementation:
- Created SysrolesFingerprint parser in insights/parsers/sysroles_fingerprint.py
  - Parses JSONL format (one JSON object per line)
  - Stores each record as a dictionary in a list
  - Handles empty files (raises SkipComponent)
  - Handles invalid JSON (raises ParseException)
  - Skips blank lines and empty JSON objects gracefully
  - Includes comprehensive docstring with realistic examples

Tests:
- Created insights/tests/parsers/test_sysroles_fingerprint.py with 12 comprehensive test cases
  - Multiple records parsing
  - Single record parsing
  - Automation Hub collection path detection
  - Empty file handling
  - Blank lines ignored
  - Whitespace-only content
  - Invalid JSON detection
  - Empty JSON objects skipped
  - Complex nested data structures
  - Documentation examples validation
- Coverage: 100% (exceeds ≥95% requirement)
- All tests passing: 12/12

Testing Instructions

- Run unit tests:
`$ pytest insights/tests/parsers/test_sysroles_fingerprint.py -v
`

- Check coverage:
`$ pytest insights/tests/parsers/test_sysroles_fingerprint.py --cov=insights.parsers.sysroles_fingerprint --cov-report=term-missing`

## Summary by Sourcery

Add a spec and parser for collecting and processing RHEL System Roles execution fingerprint telemetry.

New Features:
- Add collection of RHEL System Roles execution fingerprints from `/var/log/sysroles.jsonl`.
- Parse newline-delimited role telemetry records into preserved dictionaries for downstream analysis.

Enhancements:
- Handle blank lines, empty records, empty files, and malformed or non-object JSON inputs with appropriate parser behavior.

Documentation:
- Add documentation catalog coverage and parser usage examples for System Roles fingerprints.

Tests:
- Add comprehensive parser tests covering valid records, malformed input, edge cases, nested data, and documentation examples.

### CODE REVIEW

File: insights/parsers/sysroles_fingerprint.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        for line in content:
+            line = line.strip()
+            if line:
+                try:
+                    line_json = json.loads(line)
+                except Exception:
+                    raise ParseException("Invalid JSON line: {0}".format(line))
+                if line_json:
+                    self.append(line_json)
```
[the remark this answers is not in the document]
Comment (sabhasin) [reply insights/parsers/sysroles_fingerprint.py:68]: Applied the suggestion

File: insights/specs/__init__.py [hunk 2]
```diff
@@ -873,6 +873,7 @@ class Specs(SpecSet):
     sysctl_conf_initramfs = RegistryPoint(multi_output=True)
     sysctl_d_conf_etc = RegistryPoint(multi_output=True)
     sysctl_d_conf_usr = RegistryPoint(multi_output=True)
+    sysroles_fingerprint = RegistryPoint()
```
Comment (qinpingli) [inline insights/specs/__init__.py]: @sabhasin is obfuscation is needed for this spec?

File: insights/parsers/sysroles_fingerprint.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+"""
+Sysroles Fingerprint - file ``/var/log/sysroles.jsonl``
+========================================================
+
+This module provides a parser for the RHEL System Roles fingerprint data
+stored in ``/var/log/sysroles.jsonl``. The file is in JSONL (newline-delimited JSON)
+format, where each line contains a separate JSON object representing a system role
+execution fingerprint.
+"""
```
Comment (qinpingli) [inline insights/parsers/sysroles_fingerprint.py:9]: Could you add this new parser to `docs`? So, the developers can see it in https://insights-core.readthedocs.io/en/latest/ when this PR is merged.

File: insights/parsers/sysroles_fingerprint.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+        for line in content:
+            line = line.strip()
+            if line:
+                try:
+                    line_json = json.loads(line)
+                except Exception:
+                    raise ParseException("Invalid JSON line: {0}".format(line))
+                if not isinstance(line_json, dict):
+                    raise ParseException("Invalid JSON line: {0}".format(line))
```
Comment (qinpingli) [inline insights/parsers/sysroles_fingerprint.py:67]: Hmm, for this part. How about creating a named tuple here, such as `SYS_ROLE_FP`, if the line_json can not be convert to this named tuple, raise exception?

File: insights/specs/__init__.py (hunk 2 above)
Comment (sabhasin) [reply insights/specs/__init__.py]: No obfuscation needed :

- Data contains only standard role execution metadata (timestamps, role names, versions, counts).
- No PII: No hostnames, IPs, MACs, usernames, or secrets.

File: insights/parsers/sysroles_fingerprint.py (hunk 3 above)
Comment (sabhasin) [reply insights/parsers/sysroles_fingerprint.py:9]: Thanks for the heads-up. I have added in the Doc.

File: insights/parsers/sysroles_fingerprint.py (hunk 4 above)
Comment (sabhasin) [reply insights/parsers/sysroles_fingerprint.py:67]: Implemented dataclass approach with optional fields.

File: insights/specs/__init__.py (hunk 2 above)
Comment (qinpingli) [reply insights/specs/__init__.py]: Could you update the RegistryPoint to `sysroles_fingerprint = RegistryPoint(no_obfuscate=['hostname', 'ipv4', 'ipv6', 'mac'])`?

File: insights/parsers/sysroles_fingerprint.py (hunk 4 above)
Comment (qinpingli) [reply insights/parsers/sysroles_fingerprint.py:67]: Thanks for the update! 

From your update, seems there is no specification for this data. So I did some research, is [this](https://galaxy.ansible.com/ui/repo/published/fedora/linux_system_roles/content/module/sr_fingerprint/?version=2.4.2) the correct place? From this document, I saw the changes are big, seems it's hard to say which parameters are required ones.

If it's yes, sorry for my wrong comment, it looks like not suitable to create a data structure for it.

File: insights/specs/__init__.py (hunk 2 above)
Comment (sabhasin) [reply insights/specs/__init__.py]: Thanks for pointing it out. I have updated the RegistryPoint.

File: insights/parsers/sysroles_fingerprint.py (hunk 4 above)
Comment (sabhasin) [reply insights/parsers/sysroles_fingerprint.py:67]: Suggested changes has been made.

**Review by qinpingli — APPROVED** [review]
